test/connectivity/README.md

```diff
@@ -1,3 +1,58 @@
 # Connectivity tests
 
+![Connectivity Test Diagram](./images/ConnectivityTestDiagram.png)
 
+### Test Scenarios
+These are test scenarios to be covered in Connectivity test
+
+1. Messaging Test
+   1. Module-to-Module
+   2. Module-to-IoT Hub
+2. Twin Property Update Test
+   1. Desired property
+   2. Reported property
+3. Direct Method Test
+   1. Cloud-to-module
+   2. Ping method in EdgeAgent
+4. EdgeAgent Deployment Test
+   1. Deployment change
+
+### General flow
+1. Connectivity test can be run by calling [Connectivity Test script](https://github.com/Azure/iotedge/blob/master/test/connectivity/scripts/connectivityTest.sh).
+2. Each test will be run using AMQP and MQTT TCP protocol.
+3. For each run, a new tracking id (Guid) will be generated by test script.
+3. Test will be started after pre-defined warm-up period (default 2 minutes), which assume EdgeHub and other modules containers are up and running.
+4. Test duration is configurable, which default is 1 hour.
+5. Each module needs to tell Test Result Coordinator module the result of test operation, e.g. success/fail to make direct method calls.
+6. When test operation result reports to Test Result Coordinator, it will persist to storage after validation, e.g. supported result type.
+7. Test Result Coordinator will consume events from IoT hub's built-in event hub endpoint.  To reduce total number of messages from event hub endpoint, it is recommended to run connectivity test on its own IoT hub.
+8. Network Controller is used to mimic different network situation, e.g. offline and packet loss.  Network Controller should also report.
+9. Test report will be generated and uploaded to Log Analytics after test duration plus buffer.  The buffer is 15 minutes by default, which assumes all messages received on IoT hub are pulled for verification.
+10. After test duration, test modules should stop calling operations, e.g. sending messages, makng Direct method calls, etc.
+
+### Offline Scenario
+1. Network controller module is used to control virtual network "Azure-IoT-Edge" to mimic offline and packet loss scenarios.
+2. For 1st phase, offline scenario will be covered only;  Bandwidth throttling/Packet loss will be covered later.
+3. Network controller also need to report to Test Result Coordinator module when any network-related operation is executed.  This will impact how test operation result is verified as passed or not.
```

Comment:
I don't think you need this point here (since we already mention it above in the online case). But if you want you can almost copy it down here.